docs/csharp/language-reference/builtin-types/reference-types.md

```diff
@@ -0,0 +1,178 @@
+
+```csharp
+string str = "test";
+char x = str[2];  // x = 's';
+```
+
+In similar fashion, the `[]` operator can also be used for iterating over each character in a `string`:
+
+```csharp
+string str = "test";
+
+for (int i = 0; i < str.Length; i++)
+{
+  Console.Write(str[i] + " ");
+}
+// Output: t e s t
+``` 
+
+String literals are of type `string` and can be written in two forms, quoted and `@`-quoted. Quoted string literals are enclosed in double quotation marks ("):
+
+```csharp
+"good morning"  // a string literal
+```
+
+String literals can contain any character literal. Escape sequences are included. The following example uses escape sequence `\\` for backslash, `\u0066` for the letter f, and `\n` for newline.
+
+```csharp
+string a = "\\\u0066\n";
+Console.WriteLine(a);
+```
+
+> [!NOTE]
+> The escape code `\udddd` (where `dddd` is a four-digit number) represents the Unicode character U+`dddd`. Eight-digit Unicode escape codes are also recognized: `\Udddddddd`.
+
+[Verbatim string literals](../tokens/verbatim.md) start with `@` and are also enclosed in double quotation marks. For example:
+
+```csharp
+@"good morning"  // a string literal
+```
+
+The advantage of verbatim strings is that escape sequences are *not* processed, which makes it easy to write, for example, a fully qualified Windows file name:
+
+```csharp
+@"c:\Docs\Source\a.txt"  // rather than "c:\\Docs\\Source\\a.txt"
+```
+
+To include a double quotation mark in an @-quoted string, double it:
+
+```csharp
+@"""Ahoy!"" cried the captain." // "Ahoy!" cried the captain.
+```
+
+## The delegate type
+
+The declaration of a delegate type is similar to a method signature. It has a return value and any number of parameters of any type:
+
+```csharp
+public delegate void TestDelegate(string message);
+public delegate int TestDelegate(MyType m, long num);
+```
+
+A `delegate` is a reference type that can be used to encapsulate a named or an anonymous method. Delegates are similar to function pointers in C++; however, delegates are type-safe and secure. For applications of delegates, see [Delegates](../../programming-guide/delegates/index.md) and [Generic Delegates](../../programming-guide/generics/generic-delegates.md). Delegates are the basis for [Events](../../programming-guide/events/index.md). A delegate can be instantiated by associating it either with a named or anonymous method.
+
+The delegate must be instantiated with a method or lambda expression that has a compatible return type and input parameters. For more information on the degree of variance that is allowed in the method signature, see [Variance in Delegates](../../programming-guide/concepts/covariance-contravariance/using-variance-in-delegates.md). For use with anonymous methods, the delegate and the code to be associated with it are declared together. Both ways of instantiating delegates are discussed in this section.
```

Comment:
The last sentence implies that instantiation is discussed here, but the content really is a mention rather than a discussion.